libtenzir/src/tql2/parser.cpp

```diff
@@ -615,13 +615,12 @@ class parser {
     }
     if (const auto token = accept(tk::string)) {
       auto result = std::string{};
-      // TODO: Implement this properly.
       TENZIR_ASSERT(token.text.size() >= 2);
       TENZIR_ASSERT(token.text.front() == '"');
       TENZIR_ASSERT(token.text.back() == '"');
-      auto f = token.text.begin() + 1;
-      auto e = token.text.end() - 1;
-      for (auto it = f; it != e; ++it) {
+      const auto* f = token.text.begin() + 1;
+      const auto* e = token.text.end() - 1;
+      for (const auto* it = f; it != e; ++it) {
```

Comment:
Fwiw, this is not guaranteed to be a pointer by the standard. One of the reasons why I really dislike this lint. Same with `auto const&`… It's also just not the business of the callsite to know this.

Reply:
I know, but in practice it's actually rather nice to know that we're doing pointer arithmetic here.

Reply:
At the end of the day, it doesn't matter to me whether it's pointer arithmetic or iterator arithmetic.